app/controllers/cloud_volume_controller.rb

```diff
@@ -38,16 +38,24 @@ def specific_buttons(pressed)
     when 'cloud_volume_edit'
       javascript_redirect(:action => 'edit', :id => checked_item_id)
     when 'cloud_volume_snapshot_create'
-      javascript_redirect(:action => 'snapshot_new', :id => checked_item_id)
+      validate_results = validate_item_supports_action_button(:snapshot_create, CloudVolume)
+      if validate_results[:action_supported] then javascript_redirect(:action => 'snapshot_new', :id => checked_item_id) end
```

Comment:
can you change this line and line 47, line 50 to `javascript_redirect(:action => 'snapshot_new', :id => checked_item_id) if validate_results[:action_supported]`

Reply:
@h-kataria done